app/role_matchers.py

```diff
@@ -62,6 +62,12 @@ def invalid_role_id(role_id):
     logger.warn(msg, role_id=role_id)
     raise HTTPInternalServerError(reason=msg)
 
+def download_permission(role_id):
+    download_permission_regex = re.compile('DT-SUP.-..-..')
```

Comment:
It's probably not a good idea to `re.compile` inside a frequently-called function. The cost is minor for this regex, but it's better stored in a constant variable.

Perhaps a new `fsss_supervisors_regex` beneath the `fsss_regex`.

Otherwise, a `download_permission_regex` two lines down from the `logi_combined_regex` to ensure it's only built once.

Reply:
Added download_permission_regex below logi_combined_regex